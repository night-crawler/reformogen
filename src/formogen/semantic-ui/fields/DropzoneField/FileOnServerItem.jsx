import React from 'react';
import PropTypes from 'prop-types';
import { List, Image, Button } from 'semantic-ui-react';

import { splitExt } from '../../../utils';
import { fileTypeImageMapping, UNKNOWN_FILE_TYPE } from '../../../fileTypeImageMapping';



FileOnServerItem.propTypes = {
  fileUrl: PropTypes.string,
  getPreviewImage: PropTypes.func,
  onDelete: PropTypes.func,
  isRemovable: PropTypes.bool,
};
FileOnServerItem.defaultProps = {
  getPreviewImage: fileUrl => 
    fileTypeImageMapping[ splitExt(fileUrl)[1] ] || UNKNOWN_FILE_TYPE,
  onDelete: () => // eslint-disable-next-line
    console.warn('FileOnServerItem.onDelete'),
};
export function FileOnServerItem(props) {
  const fileName = props.fileUrl.split('/').pop();

  // TODO: watermelon come here, motherfucker!
  return (
    <List.Item>
      <Image 
        src={ props.getPreviewImage(props.fileUrl) } 
        verticalAlign='middle' size='mini'
      />
      <a href={ props.fileUrl }>{ fileName }</a>
      
      { props.isRemovable && <Button 
        icon='remove' size='mini' attached='right' floated='right' 
        onClick={ props.onDelete }
      /> }
    </List.Item>
  );
}
