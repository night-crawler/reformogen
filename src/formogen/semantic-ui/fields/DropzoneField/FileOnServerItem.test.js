import { FileOnServerItem } from './FileOnServerItem';

it('<FileOnServerItem />', () => {
  const wrapper = shallow(
    <FileOnServerItem
      fileUrl='http://sample.tld/1/2/3'
      isRemovable={ true }
    />
  );
  expect(wrapper).toMatchSnapshot();
});
