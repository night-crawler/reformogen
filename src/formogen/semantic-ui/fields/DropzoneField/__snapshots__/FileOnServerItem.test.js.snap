// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileOnServerItem /> 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FileOnServerItem
    fileUrl="http://sample.tld/1/2/3"
    getPreviewImage={[Function]}
    isRemovable={true}
    onDelete={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Image
          as="img"
          size="mini"
          src="test-file-stub"
          ui={true}
          verticalAlign="middle"
        />,
        <a
          href="http://sample.tld/1/2/3"
        >
          3
        </a>,
        <Button
          as="a"
          attached="right"
          floated="right"
          icon="remove"
          onClick={[Function]}
          role="button"
          size="mini"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "img",
          "size": "mini",
          "src": "test-file-stub",
          "ui": true,
          "verticalAlign": "middle",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "3",
          "href": "http://sample.tld/1/2/3",
        },
        "ref": null,
        "rendered": "3",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "a",
          "attached": "right",
          "floated": "right",
          "icon": "remove",
          "onClick": [Function],
          "role": "button",
          "size": "mini",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Image
            as="img"
            size="mini"
            src="test-file-stub"
            ui={true}
            verticalAlign="middle"
          />,
          <a
            href="http://sample.tld/1/2/3"
          >
            3
          </a>,
          <Button
            as="a"
            attached="right"
            floated="right"
            icon="remove"
            onClick={[Function]}
            role="button"
            size="mini"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "img",
            "size": "mini",
            "src": "test-file-stub",
            "ui": true,
            "verticalAlign": "middle",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "3",
            "href": "http://sample.tld/1/2/3",
          },
          "ref": null,
          "rendered": "3",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "a",
            "attached": "right",
            "floated": "right",
            "icon": "remove",
            "onClick": [Function],
            "role": "button",
            "size": "mini",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
